frontend/src/pages/operation_show/evidence_modals/index.tsx

```diff
@@ -145,53 +153,88 @@ export const EditEvidenceModal = (props: {
   const tagsField = useFormField<Array<Tag>>(props.evidence.tags)
   const codeblockField = useFormField<CodeBlock>({ type: 'codeblock', language: '', code: '', source: null })
   const adjustedAtField = useFormField<Date | undefined>(props.evidence.adjustedAt ?? undefined)
-  React.useEffect(() => {
+  const c2EventField = useFormField<C2Event>({type: 'c2-event', c2: '', c2Operator: '', beacon: '', externalIP: '', internalIP: '', hostname: '', userContext: '', integrity: '', processName: '', processID: 0, command: '', result: ''})
+  if (props.evidence.contentType === 'codeblock') {
+    React.useEffect(() => {
+      getEvidenceAsCodeblock({
+        operationSlug: props.operationSlug,
+        evidenceUuid: props.evidence.uuid,
+      }).then(codeblockField.onChange)
+    }, [props.evidence.contentType, codeblockField.onChange, props.operationSlug, props.evidence.uuid])
```

Comment:
It's been a bit since I last did regular work in react, but I believe that we still want all of the hooks to always execute -- this shouldn't be in a conditional.

Reply:
I had this quite messed up in the original commit. I've fixed the useEffect issue and cleaned up the return logic.